src/ws/cockpitwebservice.c

```diff
@@ -1848,6 +1867,29 @@ cockpit_web_service_get_checksum (CockpitWebService *self,
   return session ? session->checksum : NULL;
 }
 
+/**
+ * cockpit_web_service_get_transport_os_release:
+ * @self: the service
+ *
+ * Only valid after an init message has been received.
+ *
+ *  * Returns: (transfer none): a GHashTable with a key for
+ * and value of the os release data from the server transport.
+ */
+GHashTable *
+cockpit_web_service_get_transport_os_release (CockpitWebService *self,
+                                              CockpitTransport *transport)
+{
+  CockpitSession *session;
+
+  g_return_val_if_fail (COCKPIT_IS_WEB_SERVICE (self), NULL);
```

Comment:
I would suggest:
 * Caching the calculated roots in the caller
 * Generating this information on the fly
 * Caching the entire init message here.

The last point allows us to later build off this functionality around what comes in the init message.